docker/jenkins1-centos/contrib/kube-slave-common.sh

```diff
@@ -0,0 +1,171 @@
+export PATH=$PATH:${JENKINS_HOME}/.local/bin
+
+function has_service_account() {
+  [ -f "${AUTH_TOKEN}" ]
+}
+
+if has_service_account; then
+  export oc_auth="--token=$(cat $AUTH_TOKEN) --certificate-authority=${KUBE_CA}"
+  export oc_cmd="oc --server=$OPENSHIFT_API_URL ${oc_auth}"
+  export oc_serviceaccount_name="$(expr "$(oc whoami)" : 'system:serviceaccount:\w\+:\(\w\+\)' || true)"
+fi
+
+# get_imagestream_names returns a list of image streams that match the
+# SLAVE_LABEL
+function get_is_names() {
+  [ -z "$oc_cmd" ] && return
+  $oc_cmd get is -n "${PROJECT_NAME}" -l role=${SLAVE_LABEL} -o template --template "{{range .items}}{{.metadata.name}} {{end}}"
+}
+
+# convert_is_to_slave converts the OpenShift imagestream to a Jenkins Kubernetes
+# Plugin slave configuration.
+function convert_is_to_slave() {
+  [ -z "$oc_cmd" ] && return
+  local name=$1
+  local template_file=$(mktemp)
+  local template="
+  <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+    <name>{{.metadata.name}}</name>
+    <image>{{.status.dockerImageRepository}}</image>
+    <privileged>false</privileged>
+    <command></command>
+    <args></args>
+    <instanceCap>5</instanceCap>
+    <volumes/>
+    <envVars/>
+    <nodeSelector/>
+    <serviceAccount>${oc_serviceaccount_name}</serviceAccount>
+    <remoteFs>{{if index .metadata.annotations \"slave-directory\"}}{{index .metadata.annotations \"slave-directory\"}}{{else}}${DEFAULT_SLAVE_DIRECTORY}{{end}}</remoteFs>
+    <label>{{if index .metadata.annotations \"slave-label\"}}{{index .metadata.annotations \"slave-label\"}}{{else}}${name}{{end}}</label>
+  </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+  "
+  echo "${template}" > ${template_file}
+  $oc_cmd get -n "${PROJECT_NAME}" is/${name} -o templatefile --template ${template_file}
+  rm -f ${template_file} &>/dev/null
+}
+
+# generate_kubernetes_config generates a configuration for the kubernetes plugin
+function generate_kubernetes_config() {
+    [ -z "$oc_cmd" ] && return
+    local slave_templates=""
+    if has_service_account; then
+      for name in $(get_is_names); do
+        slave_templates+=$(convert_is_to_slave ${name})
+      done
+    else
+      return
+    fi
+    echo "
+    <org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud>
+      <name>openshift</name>
+      <templates>
+        <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+          <name>maven</name>
+          <image>${MAVEN_SLAVE}</image>
+          <privileged>false</privileged>
+          <command></command>
+          <args></args>
+          <instanceCap>2147483647</instanceCap>
+          <label>maven</label>
+          <volumes/>
+          <envVars/>
+          <nodeSelector/>
+          <remoteFs>/tmp</remoteFs>
+          <serviceAccount>${oc_serviceaccount_name}</serviceAccount>
+        </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+        <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+          <name>android</name>
```

Comment:
So this defines `node (android)` in the pipeline, spins up an 'Android" container/slave ?